treasury-fees-monitor/data-collection/collect-fee-data.js

```diff
@@ -0,0 +1,183 @@
+const commandLineArgs = require('command-line-args');
+const createCsvWriter = require('csv-writer').createObjectCsvWriter;
+const ethers = require('ethers');
+const fs = require('fs');
+
+require('dotenv').config();
+
+const BLOCK_CHUNK = 128;
+const SECONDS_IN_DAY = 24 * 60 * 60;
+
+async function getBlockByTimestamp(provider, userTimestamp) {
+  // seed the algorithm
+  // starting value for low block
+  let lowBlockNumber = 1;
+  let lowBlock = await provider.getBlock(lowBlockNumber);
+  let lowBlockTimestamp = lowBlock.timestamp;
+
+  // starting value for high block
+  let highBlockNumber = await provider.getBlockNumber();
+  let highBlock = await provider.getBlock(highBlockNumber);
+  let highBlockTimestamp = highBlock.timestamp;
+
+  // iterate until we find the two blocks whose timestamps bound the requested timestamp
+  let blockPerTime = 0;
+  let newBlock;
+  let newBlockNumber;
+  let newBlockTimestamp;
+  while (highBlockNumber - lowBlockNumber > 1 && lowBlockTimestamp !== newBlockTimestamp) {
+    // calculate average number of blocks per second
+    blockPerTime = (highBlockNumber - lowBlockNumber) / (highBlockTimestamp - lowBlockTimestamp);
+
+    // estimate the correct block number where the user's timestamp should exist
+    newBlockNumber = Math.floor(
+      lowBlockNumber + (blockPerTime * (userTimestamp - lowBlockTimestamp)),
+    );
+
+    // retrieve the block and timestamp
+    // eslint-disable-next-line no-await-in-loop
+    newBlock = await provider.getBlock(newBlockNumber);
+    newBlockTimestamp = newBlock.timestamp;
+
+    if (newBlockTimestamp > userTimestamp) {
+      // if the retrieved timestamp is higher than we expected, shift the high block down
+      highBlock = newBlock;
+      highBlockTimestamp = newBlockTimestamp;
+      highBlockNumber = newBlockNumber;
+    } else if (newBlockTimestamp < userTimestamp) {
+      // if the retrieved timestamp is lower than we expected, shift the high block down
+      lowBlock = newBlock;
+      lowBlockTimestamp = newBlockTimestamp;
+      lowBlockNumber = newBlockNumber;
+    } else {
+      // we found the timestamp exactly (this is unlikely, but we need to check)
+      return newBlock;
+    }
+  }
+
+  // the low block number should now have the highest timestamp that does not exceed the user
+  // requested timestamp
+  return lowBlock;
+}
+
+async function collectData(provider, filterInfo, timePeriod) {
+  const endBlockNum = await provider.getBlockNumber();
+  const endBlock = await provider.getBlock(endBlockNum);
+
+  const startTimestamp = (endBlock.timestamp - (timePeriod * SECONDS_IN_DAY));
+  const startBlock = await getBlockByTimestamp(provider, startTimestamp);
+  const startBlockNum = startBlock.number;
+
+  // chunk each getLogs call into a smaller block range so there is less of a chance that a
+  // 'too many logs' type error occurs
+  const promises = [];
+  for (let i = startBlockNum; i <= endBlockNum; i += BLOCK_CHUNK) {
```

Comment:
I _think_ that `eth_getLogs` will perform a request inclusive of the first and last block numbers requested.  If that is true, then the increment needs to be one higher than the block size, otherwise you would end up with a one block overlap between requests:
1 - 101
101 - 201
etc.

```suggestion
  for (let i = startBlockNum; i <= endBlockNum; i += BLOCK_CHUNK + 1) {
```